### PR TITLE
Travis: use flake8 instead of pycodestyle

### PR DESCRIPTION
**Pull request by astrofrog**
At the moment I haven't added any new warnings/errors to check for, but with this we can start checking for pyflakes errors by adding them to the list in the flake8 section of setup.cfg. Note that this requires astropy/ci-helpers#158 to be merged first.

### CODE REVIEW

**Comment by bsipocz** [comment]
@astrofrog - Are the error/warning codes the same? 

Travis can be restarted now.

**Comment by astrofrog** [comment]
@bsipocz - yes, restarting now

**Comment by pllim** [comment]
I guess the checks are working?

```
astropy/io/ascii/tests/test_rst.py:135:18: W291 trailing whitespace

astropy/io/ascii/tests/test_rst.py:137:18: W291 trailing whitespace

astropy/io/ascii/tests/test_rst.py:140:18: W291 trailing whitespace
```

**Comment by astrofrog** [comment]
Note that this doesn't close https://github.com/astropy/astropy/pull/5710 because https://github.com/astropy/astropy/pull/5710 is still relevant if people want to run pycodestyle in the repo.

**Comment by bsipocz** [comment]
@astrofrog - could you edit the description then? otherwise merging this will close #5710 automatically

**Comment by pllim** [comment]
Ops. That was me. Sorry. I thought this *replaces* pycodestyle.

**Comment by bsipocz** [comment]
Thinking about this a bit more, actually does it have any function rather than for our CI to list those code selection? This case the editor highlights may be silenced for the ones not listed in there and it should be edited, too when someone runs pycodestyle to find all cases.

**Comment by astrofrog** [comment]
> Thinking about this a bit more, actually does it have any function rather than for our CI to list those code selection? This case the editor highlights may be silenced for the ones not listed in there and it should be edited, too when someone runs pycodestyle to find all cases.

Do you mean I should add a comment about this? Or should I make it so that we don't list this in setup.cfg but instead e.g. just in .travis.yml?

**Comment by bsipocz** [comment]
@astrofrog - I think the best would be to define the select list in .travis.yml, while only list an ignore list in setup.cfg for things that we definitely never will enforce.    On the other hand I see some good argument to have the list in setup.cfg, so that people can easily make sure their PR won't brake on travis. But in the case I would only keep the list for the style checker we use in travis, e.g. remove it from under pycodestyle/pep8 and only keep it for flake8.

File: astropy/io/ascii/tests/test_rst.py [hunk 1]
```diff
@@ -132,12 +132,12 @@ def test_trailing_spaces_in_row_definition():
     """ Trailing spaces in the row definition column shouldn't matter"""
     table = """
 # comment (with blank line above)
-   ==== ==== ====    
+   ==== ==== ====
```
Comment (mhvk) [inline astropy/io/ascii/tests/test_rst.py]: Given the comment above, I think the trailing spaces were there on purpose, no? I.e., really, the `nopep8` below should be changed.

**Comment by mhvk** [comment]
Just to be sure: is the purpose of this PR simply to allow users to run `flake8` and get the same result as they would if they ran `pycodestyle`? If so, the current implementation of the list in `setup.cfg` does seem the most logical.

But note that in the example, I think trailing spaces were incorrectly removed, so there needs to be a way to turn off `flake8` checking as well (or might this fit under `noqa`?).

**Comment by mhvk** [comment]
p.s. I guess the title of this PR suggests that this is specifically for travis. Why is this better than `pycodestyle`?

Also, if we do use it on `travis`, I think we should document it in the development section.

**Comment by bsipocz** [comment]
@mhvk - afaik flake is better as it check for issues that are not covered by pycodestyle (e.g. unused imports, variables, etc. if I remember correctly). It would be great to gradually add more error codes to the checks, at least those that are unambiguously a good thing to follow.

**Comment by astrofrog** [comment]
@mhvk - the main purpose of the PR is actually to use flake8 on Travis which means that we can start also include pyflakes error codes to those that need to be checked. For example we can check for unused imports or undefined variables. I haven't added those codes yet because that will require going through and fixing all instances of errors, but this lays the foundation so we can do that.

**Comment by astrofrog** [comment]
Basically, flake8 = pep8 + pyflakes

**Comment by mhvk** [comment]
Ah, great!  So, in that case, I think a general comment would be to add this to the developer documentation as well.

**Comment by bsipocz** [comment]
@astrofrog - Could you address @mhvk inline comment? I'm trying to merge the trivial 1.3.1 PRs, so we can make a release.

**Comment by bsipocz** [comment]
My https://github.com/astropy/astropy/pull/5730#issuecomment-278414529 is still relevant, but can be addressed later separately if you agree with the line of thoughts. So I would rather go ahead and merge this (failures are unrelated urlerrors).

**Pull request merged**